### PR TITLE
Allow bin.as to be an array of 2 strings

### LINKED ISSUE #3831: Allow bin's `as` to be an array of (two) strings?
Currently `as` is a string, which makes sense if we only want to used the binned output in aggregate.

However, if we plan to use the two output to do additional calculation, then it's unclear from the spec that the second field is `<field_name>_end`.  Given that we enforce all fields in `transform` to be named, this can violate our policy if `<field_name>_end` is directly used. 

cc: @domoritz @invokesus

### PR DESCRIPTION
**Pull request by invokesus**
Fixes #3831

### CODE REVIEW

File: src/compile/data/bin.ts [hunk 1]
```diff
@@ -38,15 +38,18 @@ function getSignalsFromModel(model: Model, key: string) {
   };
 }
 
+// Bad Name?
```
Comment (kanitw) [inline src/compile/data/bin.ts]: What would be a better name?

File: src/compile/data/bin.ts [hunk 2]
```diff
@@ -38,15 +38,18 @@ function getSignalsFromModel(model: Model, key: string) {
...
   return 'as' in t;
 }
 
 function createBinComponent(t: FieldDef<string> | BinTransform, model: Model) {
   let as: [string, string];
 
-  if (isBinTransform(t)) {
+  if (isBinTransform(t) && isString(t.as)) {
     as = [t.as, `${t.as}_end`];
+  } else if (isBinTransform(t)) {
```
Comment (kanitw) [inline src/compile/data/bin.ts]: I think you better add `if (isString(t.as))` in side the original `isBinTransform` if branch

File: src/compile/data/bin.ts (hunk 1 above)
Comment (invokesus) [reply src/compile/data/bin.ts]: My bad. I didn't pay attention to the type of the argument.

**Comment by kanitw** [comment]
Not sure why one of the example is outdated.

**Comment by invokesus** [comment]
Can't reproduce locally 😕

**Comment by kanitw** [comment]
Try rebasing to the latest master, upgrade deps, and re-run build:only and build:example?

File: src/compile/data/parse.ts [hunk 3]
```diff
@@ -57,9 +57,12 @@ export function parseTransformArray(head: DataFlowNode, model: Model, ancestorPa
...
       head = new FilterNode(head, model, t.filter);
     } else if (isBin(t)) {
-      head = BinNode.makeFromTransform(head, t, model);
+      const bin = head = BinNode.makeFromTransform(head, t, model);
+
+      for (const field of keys(bin.producedFields())) {
+        ancestorParse.set(field, 'number', false);
+      }
 
-      ancestorParse.set(t.as, 'number', false);
```
Comment (invokesus) [inline src/compile/data/parse.ts]: So, previously only the `as` `string` was being set. Now, both `as` elements(implicit as well as explicit) are set, which changed that example. The svg is unchanged.

**Pull request merged**